From 91925175a3701027edf62fd9c8db4c0c803112fe Mon Sep 17 00:00:00 2001
From: WofWca <wofwca@protonmail.com>
Date: Sat, 24 Jan 2026 16:56:17 +0400
Subject: [PATCH] fix: better gib velocities for dead bodies

---
 code/cgame/cg_effects.c | 22 ++++++++++++----------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/code/cgame/cg_effects.c b/code/cgame/cg_effects.c
index c82d2364..1065eaa8 100644
--- a/code/cgame/cg_effects.c
+++ b/code/cgame/cg_effects.c
@@ -676,9 +676,6 @@ void CG_GibPlayer( const vec3_t playerOrigin, const vec3_t playerAngles,
 	} else {
 		bodyAngles[PITCH] = 0;
 	}
-	// TODO fix: if `AdjustPositionIfDeathAnimation()` ran,
-	// `forward` could potentially be pointing up,
-	// so some of our velocity calculations below are not quite right.
 	AngleVectors( bodyAngles, forward, right, up );
 
 	VectorScale( playerVelocity, cg_gibsInheritPlayerVelocity.value, playerVelocityScaled );
@@ -690,14 +687,15 @@ void CG_GibPlayer( const vec3_t playerOrigin, const vec3_t playerAngles,
 		VectorCopy( baseOrigin, origin );
 		VectorMA(origin, MINS_Z + 0.95 * playerHeight, up, origin);
 		VectorClear( velocity );
-		velocity[0] = crandom()*baseRandomVelocity;
-		velocity[1] = crandom()*baseRandomVelocity;
+		VectorMA( velocity, crandom()*baseRandomVelocity, forward, velocity );
+		VectorMA( velocity, crandom()*baseRandomVelocity, right, velocity );
 		// For the skull / brain we want the random velocity
 		// to never have downwards (inwards) component,
 		// so we use `random` instead of `crandom`.
 		// We also do the same for other gibs,
 		// but for the left / right velocity components.
-		velocity[2] = jump + random()*baseRandomVelocity;
+		VectorMA( velocity, random()*baseRandomVelocity, up, velocity );
+		velocity[2] += jump;
 		VectorAdd( velocity, playerVelocityScaled, velocity );
 		if ( !skullLaunched && (rand() & 1) ) {
 			CG_LaunchGib( origin, lookDirAngles, velocity, cgs.media.gibSkull );
@@ -728,7 +726,8 @@ void CG_GibPlayer( const vec3_t playerOrigin, const vec3_t playerAngles,
 		VectorClear( velocity );
 		VectorMA( velocity, +random()*baseRandomVelocity, right, velocity );
 		VectorMA( velocity, crandom()*baseRandomVelocity, forward, velocity );
-		velocity[2] = jump + crandom()*baseRandomVelocity;
+		VectorMA( velocity, crandom()*baseRandomVelocity, up, velocity );
+		velocity[2] += jump;
 		VectorAdd( velocity, playerVelocityScaled, velocity );
 		VectorCopy( bodyAngles, angles );
 		angles[ROLL] += 70;
@@ -790,7 +789,8 @@ void CG_GibPlayer( const vec3_t playerOrigin, const vec3_t playerAngles,
 		VectorClear( velocity );
 		VectorMA( velocity, -random()*baseRandomVelocity, right, velocity );
 		VectorMA( velocity, crandom()*baseRandomVelocity, forward, velocity );
-		velocity[2] = jump + crandom()*baseRandomVelocity;
+		VectorMA( velocity, crandom()*baseRandomVelocity, up, velocity );
+		velocity[2] += jump;
 		VectorAdd( velocity, playerVelocityScaled, velocity );
 		VectorCopy( bodyAngles, angles );
 		angles[ROLL] -= 90;
@@ -819,7 +819,8 @@ void CG_GibPlayer( const vec3_t playerOrigin, const vec3_t playerAngles,
 		VectorClear( velocity );
 		VectorMA( velocity, +random()*baseRandomVelocity, right, velocity );
 		VectorMA( velocity, crandom()*baseRandomVelocity, forward, velocity );
-		velocity[2] = jump + crandom()*baseRandomVelocity;
+		VectorMA( velocity, crandom()*baseRandomVelocity, up, velocity );
+		velocity[2] += jump;
 		VectorAdd( velocity, playerVelocityScaled, velocity );
 		VectorCopy( bodyAngles, angles );
 		angles[ROLL] -= 30;
@@ -836,7 +837,8 @@ void CG_GibPlayer( const vec3_t playerOrigin, const vec3_t playerAngles,
 		VectorClear( velocity );
 		VectorMA( velocity, -random()*baseRandomVelocity, right, velocity );
 		VectorMA( velocity, crandom()*baseRandomVelocity, forward, velocity );
-		velocity[2] = jump + crandom()*baseRandomVelocity;
+		VectorMA( velocity, crandom()*baseRandomVelocity, up, velocity );
+		velocity[2] += jump;
 		VectorAdd( velocity, playerVelocityScaled, velocity );
 		VectorCopy( bodyAngles, angles );
 		angles[PITCH] += 15;
-- 
2.54.0

