From f21172d1ea09b2509c6245a7eeb7c583f2c65fcf Mon Sep 17 00:00:00 2001
From: WofWca <wofwca@protonmail.com>
Date: Tue, 17 Feb 2026 19:46:52 +0400
Subject: [PATCH] fix: shotgun not respecting `MAX_KNOCKBACK`

---
 code/game/g_combat.c | 32 ++++++++++++++++++++++++++++++--
 1 file changed, 30 insertions(+), 2 deletions(-)

diff --git a/code/game/g_combat.c b/code/game/g_combat.c
index c5b6e8e0..3e1d10b4 100644
--- a/code/game/g_combat.c
+++ b/code/game/g_combat.c
@@ -229,6 +229,7 @@ void LookAtKiller( gentity_t *self, gentity_t *inflictor, gentity_t *attacker )
 	self->client->ps.stats[STAT_DEAD_YAW] = vectoyaw ( dir );
 }
 
+#define MAX_KNOCKBACK 200
 /*
 ==================
 KnockbackToKnockbackSpeed
@@ -1001,8 +1002,8 @@ void G_Damage( gentity_t *targ, gentity_t *inflictor, gentity_t *attacker,
 	}
 
 	knockback = damage;
-	if ( knockback > 200 ) {
-		knockback = 200;
+	if ( knockback > MAX_KNOCKBACK ) {
+		knockback = MAX_KNOCKBACK;
 	}
 	if ( targ->flags & FL_NO_KNOCKBACK ) {
 		knockback = 0;
@@ -1189,6 +1190,33 @@ void G_Damage( gentity_t *targ, gentity_t *inflictor, gentity_t *attacker,
 					knockback, kvel, dflags, mod, velChange );
 				VectorAdd(targ->client->ps.velocity, velChange, targ->client->ps.velocity);
 			}
+			// If we already got to max knockback, don't apply any more of it.
+			// Otherwise one quad shotgun shot can get you 330 knockback,
+			// resuling in gibs flying too fast, much faster
+			// than from e.g. a railgun shot.
+			// It would make sense to do this adjustment always,
+			// but let's only apply this to shotgun gib deaths,
+			// to be closer to how the original game works.
+			//
+			// Note that we're using `damage_knockback`, which is only cleared
+			// in `CliendEndFrame` and not immediately after the shot.
+			// But it's good enough since this is basically only about
+			// gib and camera speed.
+			if (
+				targ->health <= GIB_HEALTH && g_blood.integer &&
+				!g_oldGibs.integer &&
+				mod == MOD_SHOTGUN &&
+				targ->client && targ->client->damage_knockback > MAX_KNOCKBACK ) {
+				int excess = targ->client->damage_knockback - MAX_KNOCKBACK;
+				if ( excess > knockback ) {
+					// It's excess knockback from this particular shot,
+					// not total excess knockback.
+					excess = knockback;
+				}
+				VectorMA( targ->client->ps.velocity,
+					KnockbackToKnockbackSpeed( -excess ), dir,
+					targ->client->ps.velocity );
+			}
 
 			targ->enemy = attacker;
 			targ->die (targ, inflictor, attacker, take, mod);
-- 
2.54.0

