From e6fdded193a0dfdda85d33485817bff9d7fe3cbd Mon Sep 17 00:00:00 2001
From: WofWca <wofwca@protonmail.com>
Date: Sat, 9 May 2026 13:47:52 +0400
Subject: [PATCH] fix: gib random speed after match end being lower

---
 code/game/g_combat.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/code/game/g_combat.c b/code/game/g_combat.c
index 48b55392..d5549dc6 100644
--- a/code/game/g_combat.c
+++ b/code/game/g_combat.c
@@ -278,7 +278,7 @@ void GibEntity( gentity_t *self, int killer, const int damageBloodFallback ) {
 	// we're simply providing the client with the knockback info,
 	// and whether to use that into is up to `cg_oldGibs`.
 	if ( g_gibsNewEvGibPlayerParmProtocol.integer == 1 ) {
-		int damage;
+		int damage, frameDamage;
 		float knockbackSpeed;
 
 		// We prefer actual damage over `client->damage_knockback`
@@ -291,8 +291,14 @@ void GibEntity( gentity_t *self, int killer, const int damageBloodFallback ) {
 		// "no knockback" means "the player should not be moved
 		// in any particular direction",
 		// and not that "their gibs should stay put".
-		damage = self->client
-			? self->client->damage_blood + self->client->damage_armor
+		//
+		// In case when `level.intermissionQueued` we don't use
+		// `damage_blood` and `damage_armor` because they would not be set.
+		// See `level.intermissionQueued` and `targ->die == body_die` check
+		// in `G_Damage`.
+		frameDamage = self->client->damage_blood + self->client->damage_armor;
+		damage = self->client && !( level.intermissionQueued && frameDamage == 0 )
+			? frameDamage
 			: damageBloodFallback;
 		if ( damage > MAX_KNOCKBACK ) {
 			damage = MAX_KNOCKBACK;
-- 
2.54.0

