From 3e4bbd61c16e70a9440b8f540086acc654594b86 Mon Sep 17 00:00:00 2001
From: WofWca <wofwca@protonmail.com>
Date: Sat, 21 Feb 2026 23:11:14 +0400
Subject: [PATCH] refactor: simplify `cg_event.c` EV_GIB a little

---
 code/cgame/cg_event.c | 23 +++++++++--------------
 1 file changed, 9 insertions(+), 14 deletions(-)

diff --git a/code/cgame/cg_event.c b/code/cgame/cg_event.c
index 635a09934..442337277 100644
--- a/code/cgame/cg_event.c
+++ b/code/cgame/cg_event.c
@@ -1227,6 +1227,13 @@ void CG_EntityEvent( centity_t *cent, vec3_t position ) {
 				// Just use the default knockback speed for 100 damage.
 				: 100 * 1000 / COMBAT_PLAYER_MASS;
 
+			// Apparently at this point `es->pos.trDelta` doesn't yet have
+			// the knockback from the damage that gibbed us,
+			// so we have to differentiate between self and non-self.
+			vec3_t *vel = es->number == cg.snap->ps.clientNum
+				? &cg.predictedPlayerState.velocity
+				: &es->pos.trDelta;
+
 			lerpFrame_t torsoAnimation = es->number == cg.snap->ps.clientNum
 				// `cent->pe.torso` appears to be not good for self.
 				? cg.predictedPlayerEntity.pe.torso
@@ -1262,20 +1269,8 @@ void CG_EntityEvent( centity_t *cent, vec3_t position ) {
 			torsoAngles[YAW] = torsoAnimation.yawAngle;
 			torsoAngles[ROLL] = 0;
 
-			if ( es->number == cg.snap->ps.clientNum ) {
-				// Apparently at this point `es->pos.trDelta` doesn't yet have
-				// the knockback from the damage that gibbed us,
-				// so we have to differentiate between self and non-self,
-				// and use `cg.predictedPlayerState.velocity`
-				// if it's ourself.
-				CG_GibPlayer( cent->lerpOrigin, torsoAngles,
-					cg.predictedPlayerState.velocity, knockbackSpeed,
-					&torsoAnimation, randSeed );
-			} else {
-				CG_GibPlayer( cent->lerpOrigin, torsoAngles,
-					es->pos.trDelta, knockbackSpeed,
-					&torsoAnimation, randSeed );
-			}
+			CG_GibPlayer( cent->lerpOrigin, torsoAngles, *vel, knockbackSpeed,
+				&torsoAnimation, randSeed );
 		}
 		break;
 
-- 
2.55.0

