From c986961fd9fa863e4727342b61fcafa96da100e1 Mon Sep 17 00:00:00 2001
From: WofWca <wofwca@protonmail.com>
Date: Tue, 30 Jun 2026 17:04:33 +0400
Subject: [PATCH] fix: better dead bodies' explosion knockback dir

---
 code/game/g_combat.c | 21 ++++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/code/game/g_combat.c b/code/game/g_combat.c
index 99fd0514..80ded494 100644
--- a/code/game/g_combat.c
+++ b/code/game/g_combat.c
@@ -1602,9 +1602,24 @@ qboolean G_RadiusDamage ( vec3_t origin, gentity_t *attacker, float damage, floa
 				hitClient = qtrue;
 			}
 			VectorSubtract (ent->r.currentOrigin, origin, dir);
-			// push the center of mass higher than the origin so players
-			// get knocked into the air more
-			dir[2] += 24;
+			if ( ent->health > 0 || g_oldGibs.integer ) {
+				// push the center of mass higher than the origin so players
+				// get knocked into the air more
+				dir[2] += 24;
+
+				// Also results in `+= 24`, for live players
+				// with default `mins` and `maxs`,
+				// but let's keep the old calculation to be 100% sure
+				// that gameplay is the same as in vanilla.
+				//
+				// dir[2] += ent->r.mins[2] +
+				// 	( ent->r.maxs[2] - ent->r.mins[2] ) * 6/7;
+			} else {
+				// For dead bodies prefer "realism":
+				// use the middle of the bounding box as the center of mass.
+				dir[2] += ent->r.mins[2] +
+					( ent->r.maxs[2] - ent->r.mins[2] ) * 1/2;
+			}
 			G_Damage (ent, NULL, attacker, dir, origin, (int)points, DAMAGE_RADIUS, mod);
 		}
 	}
-- 
2.55.0

